src/Functions/array/indexOfAssumeSorted.cpp

```diff
@@ -13,16 +13,27 @@ using FunctionIndexOfAssumeSorted = FunctionArrayIndex<IndexOfAssumeSorted, Name
 
 REGISTER_FUNCTION(IndexOfAssumeSorted)
 {
-    factory.registerFunction<FunctionIndexOfAssumeSorted>(FunctionDocumentation{
-        .description = R"(
-The function finds the position of the first occurrence of element X in the array.
-Indexing from one.
-The function can be used when the internal array type is not Nullable and the array is sorted in non-decreasing order.
-If the array type is Nullable, the 'indexOf' function will be used.
-The binary search algorithm is used for the search.
-For more details, see [https://en.wikipedia.org/wiki/Binary_search]
-For an unsorted array, the behavior is undefined.
-)",
-        .examples = {{.name = "", .query = "SELECT indexOfAssumeSorted([1, 2, 2, 2, 3, 3, 3, 4], 3) FROM test_table;", .result=""}}, .category = FunctionDocumentation::Category::Array});
+    FunctionDocumentation::Description description = R"(
+Returns the index of the first element with value 'x' (starting from `1`) if it is in the array.
+If the array does not contain the searched-for value, the function returns `0`.
+
+:::note
+Unlike the `indexOf` function, this function assumes that the array is sorted in
+ascending order. If the array is not sorted, results are undefined.
+If the internal array is of type Nullable, then function `indexOf` will be called.
+:::
+    )";
+    FunctionDocumentation::Syntax syntax = "indexOfAssumeSorted(arr, x)";
+    FunctionDocumentation::Arguments arguments = {
+        {"arr", "A sorted array to search. [`Array`](/sql-reference/data-types/array)."},
```

Comment:
I guess `Array(T)` would be more precise as a type. Same in other places in this PR.